client/pyscripts/fm11rf08_full.py

```diff
@@ -0,0 +1,942 @@
+# Patch keys in to data
+# >> "key[][]"
+# >> "data[]"
+# >> keyok!
+'''
+  3 | 00 00 00 00 00 00 87 87 87 69 00 00 00 00 00 00 | .........i......
+'''
+#==============================================================================
+def patchKeys(keyok):
+	global  key
+	global  data
+
+	lprint(prompt)
+	lprint(f"{prompt} Patch keys in to data")
+
+	for sec in range(0, 16+1):
+		blk = (sec * 4) +3  # find "trailer" for this sector
+		if keyok:
+			if key[sec][0] == "":
+				keyA = "-- -- -- -- -- -- "
+			else:
+				kstr = key[sec][0].hex()
+				keyA = "".join([kstr[i:i+2] + " " for i in range(0, len(kstr), 2)])
+
+			if key[sec][1] == "":
+				keyB = "-- -- -- -- -- -- "
+			else:
+				kstr = key[sec][1].hex()
+				keyB = "".join([kstr[i:i+2] + " " for i in range(0, len(kstr), 2)])
+
+			data[blk] = data[blk][:6] + keyA + data[blk][24:36] + keyB
+
+		else:
+			data[blk] = data[blk][:6] + "-- -- -- -- -- -- " + data[blk][24:36] + "-- -- -- -- -- --"
+
+#+=============================================================================
+# Dump data
+# >> blkn
+# >> "data[]"
+#==============================================================================
+def dumpData():
+	global  blkn
+	global  data
+
+	lprint(prompt)
+	lprint(f"{prompt} ===========")
+	lprint(f"{prompt}  Card Data")
+	lprint(f"{prompt} ===========")
+	lprint(f"{prompt}")
+
+	cnt = 0
+	for n in blkn:
+		sec = (cnt // 4)
+		if sec > 15:  sec = sec + 16
+
+		if (n % 4 == 0):
+			lprint(f"{prompt} {sec:2d}:{data[cnt]}")
+		else:
+			lprint(f"{prompt}   :{data[cnt]}")
+
+		cnt += 1
+		if (cnt % 4 == 0) and (n != blkn[-1]):  # Space between sectors
+			lprint(prompt)
+
+#+=============================================================================
+# Let's try to detect a Bambu card by the date strings...
+#==============================================================================
+def detectBambu():
+	try:
+		dl  = bytes.fromhex(data[12][ 6:53]).decode('ascii').rstrip('\x00')
+		dls = dl[2:13]
+		ds  = bytes.fromhex(data[13][ 6:41]).decode('ascii').rstrip('\x00')
+	except Exception as e:
+		return False
+
+	# ds      24_03_22_16
+	# dl    2024_03_22_16_29
+	#       yy y    y     m    m     d    d     h    h     m    m
+	exp = r"20[2-3][0-9]_[0-1][0-9]_[0-3][0-9]_[0-2][0-9]_[0-5][0-9]"
+
+	lprint(f"{prompt}")
+	if re.search(exp, dl) and (ds == dls):
+		lprint(f"{prompt} Bambu date strings detected.")
+		return True
+	else:
+		lprint(f"{prompt} Bambu date strings not detected.")
+		return False
+
+
+#+=============================================================================
+# Dump bambu details
+# https://github.com/Bambu-Research-Group/RFID-Tag-Guide/blob/main/README.md
+# >> "data[]"
+'''
+      6           18          30          42         53
+      |           |           |           |          |
+  3 | 00 00 00 00 00 00 87 87 87 69 00 00 00 00 00 00 | .........i......
+'''
+#+=============================================================================
+def dumpBambu():
```

Comment:
it would be interesting to isolate this function in a separate file to be imported, so other ppl can use it in their scripts s well, or call the script with a provided block0

Reply:
qv. https://github.com/RfidResearchGroup/proxmark3/pull/2613#discussion_r1830270895